File: docs/examples/rs/examples/data_querying_timestamp.rs
use std::time::SystemTime;

use bytes::Bytes;
use reduct_rs::{ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a client instance, then get or create a bucket
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();
    let bucket = client.create_bucket("test").exist_ok(true).send().await?;

    // Send a record to the "rs-example" entry with the current timestamp
    let timestamp = SystemTime::now();
    bucket
        .write_record("rs-example")
        .timestamp(timestamp)
        .data("Some binary data")
        .send()
        .await?;

    // Read the record by timestamp
    let record = bucket
        .read_record("rs-example")
        .timestamp(timestamp)
        .send()
        .await?;

    println!("Timestamp: {:?}", record.timestamp());
    println!("Content Length: {}", record.content_length());
    println!("Content Type: {}", record.content_type());
    println!("Labels: {:?}", record.labels());

    // Read the record data
    let data = record.bytes().await?;
    assert_eq!(data, Bytes::from("Some binary data"));

    Ok(())
}

File: versioned_docs/version-1.19.x/examples/go/src/data_querying_timestamp.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	"time"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get or create a bucket with the name "my-bucket"
	bucket, err := client.CreateOrGetBucket(context.Background(), "my-bucket", nil)
	if err != nil {
		panic(err)
	}

	// Send a record to the "go-example" entry with the current timestamp
	ts := time.Now().UnixMicro()
	err = bucket.BeginWrite(context.Background(), "go-example", &reduct.WriteOptions{
		Timestamp:   ts,
		ContentType: "application/octet-stream",
	}).Write([]byte("Some binary data"))
	if err != nil {
		panic(err)
	}

	// Read the record in the "go-example" entry of the bucket
	record, err := bucket.BeginRead(context.Background(), "go-example", &ts)
	if err != nil {
		panic(err)
	}

	// Print meta information
	println("Timestamp:", record.Time())
	println("Content Length:", record.Size())
	println("Content Type:", record.ContentType())
	println("Labels:", record.Labels())

	// Read the record content
	content, err := record.ReadAsString()
	if err != nil {
		panic(err)
	}

	if string(content) != "Some binary data" {
		panic("Content mismatch")
	}
}
